File: src/main/java/com/cabservice/service/TripService.java
package com.cabservice.service;

import com.cabservice.dto.request.TripRequestDTO;
import com.cabservice.dto.response.TripResponseDTO;

import java.util.List;

public interface TripService {


    TripResponseDTO bookTrip(TripRequestDTO request);

    TripResponseDTO getMyTripById(Long tripId);

    List<TripResponseDTO> getMyTrips();

    TripResponseDTO cancelMyTrip(Long tripId);

}

File: ui/src/services/tripService.js
import api from "./api";

export const bookTrip = async (tripData) => {

	const response = await api.post(
		"/trips",
		tripData
	);

	return response.data;
};

export const getMyTrips = async () => {

	const response = await api.get(
		"/trips/my"
	);

	return response.data;
};

export const getMyTripById = async (tripId) => {

	const response = await api.get(
		`/trips/my/${tripId}`
	);

	return response.data;
};

export const cancelMyTrip = async (tripId) => {

	const response = await api.patch(
		`/trips/my/${tripId}/cancel`
	);

	return response.data;
};